qa/integration-tests/src/test/java/io/camunda/it/backup/BackupRestoreIT.java

```diff
@@ -0,0 +1,257 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.SelinuxContext;
+import org.testcontainers.shaded.org.awaitility.Awaitility;
+
+@ZeebeIntegration
+public class BackupRestoreIT {
+  private static final String REPOSITORY_NAME = "test-repository";
+  @TempDir public Path repositoryDir;
+  protected CamundaClient camundaClient;
+
+  @TestZeebe(autoStart = false)
+  protected TestStandaloneCamunda testStandaloneCamunda;
+
+  private RestClient restClient;
+  private ElasticsearchClient esClient;
+  private org.opensearch.client.RestClient osRestClient;
+  private OpenSearchClient opensearchClient;
+  //  private GenericContainer<?> storageContainer;
+  private String storagePath;
+  private BackupRestoreTestConfig config;
+
+  private void setupCamunda(final BackupRestoreTestConfig config) throws IOException {
+    //    startStorageContainer(config);
+    testStandaloneCamunda = new TestStandaloneCamunda();
+    this.config = config;
+    config.configure(testStandaloneCamunda, repositoryDir);
+    testStandaloneCamunda.start();
+    testStandaloneCamunda.awaitCompleteTopology();
+    createSearchClient();
+    createRepository(storagePath);
+  }
+
+  @AfterEach
+  public void afterEach() throws IOException {
+    testStandaloneCamunda.stop();
+    if (restClient != null) {
+      restClient.close();
+      esClient = null;
+    }
+    if (osRestClient != null) {
+      osRestClient.close();
+      opensearchClient = null;
+    }
+  }
+
+  public static Stream<BackupRestoreTestConfig> sources() {
+    return Stream.of(DatabaseType.ELASTICSEARCH)
+        .flatMap(
+            dbType ->
+                Stream.of(RepositoryType.values())
+                    .map(
+                        repositoryType ->
+                            new BackupRestoreTestConfig(dbType, repositoryType, "bucket"))
+                    .limit(1));
+  }
+
+  @ParameterizedTest
+  @MethodSource(value = {"sources"})
+  public void shouldBackupAndRestoreToPreviousState(final BackupRestoreTestConfig config)
+      throws IOException, InterruptedException {
+    // given
+    setupCamunda(config);
+    Thread.sleep(10000);
+    final var backupClient = testStandaloneCamunda.newBackupClient();
+    // a backup is requested
+    final var takeResponse = backupClient.takeBackup(1L);
+    assertThat(takeResponse)
+        .satisfies(Either::isLeft)
+        .extracting(r -> r.get().getScheduledSnapshots())
+        .asInstanceOf(InstanceOfAssertFactories.LIST)
+        .isNotEmpty();
+    final var snapshots = takeResponse.get().getScheduledSnapshots();
+    // the backup is completed
+    Awaitility.await("Backup completed")
+        .atMost(Duration.ofSeconds(600))
+        .until(
```

Comment:
:x: Please use `untilAsserted` to get the error message of the last failed assertion, versus just waiting until something is true. This makes diagnosing the test failure faster.